eyeball-im-util/src/vector/limit.rs

```diff
@@ -0,0 +1,372 @@
+use std::{
+    cmp::{min, Ordering},
+    collections::VecDeque,
+    pin::Pin,
+    task::{self, ready, Poll},
+};
+
+use super::{
+    VectorDiffContainer, VectorDiffContainerFamily, VectorDiffContainerOps,
+    VectorDiffContainerStreamElement, VectorDiffContainerStreamFamily,
+};
+use eyeball_im::VectorDiff;
+use futures_core::Stream;
+use imbl::{vector, Vector};
+use pin_project_lite::pin_project;
+
+pin_project! {
+    /// A [`VectorDiff`] stream adapter that presents a limited view of the
+    /// underlying [`ObservableVector`]s items.
+    ///
+    /// For example, be `S` a `Stream<Item = VectorDiff>`. The `Vector` represented
+    /// by `S` can have any length, but one may want to virtually _limit_ this `Vector`
+    /// to a certain size. Then this `DynamicLimit` adapter is well appropriated.
+    /// The limit is dynamic, i.e. it changes over time based on values that are polled
+    /// from another `Stream` (ref. [`Self::limit_stream`]).
+    ///
+    /// Because the limit is dynamic, an internal buffered vector is kept, so that
+    /// the adapter knows which values can be added when the limit is increased, or
+    /// when values are removed and new values must be inserted. This fact is important
+    /// if the items of the `Vector` have a non-negligeable size.
+    ///
+    /// It's OK to have a limit larger than the length of the observed `Vector`.
+    ///
+    /// Note that this adapter won't produce anything until a new limit is polled.
```

Comment:
```suggestion
    /// Note that this adapter won't produce anything until the first limit is polled.
```

Reply:
I'm still not very happy with this. Actually, given the current structure of the type, we can move this comment to the `new` constructor and provide an additional `with_initial_limit` constructor that also returns the initial limited values for chaining with other adapters, right?

Reply:
I've moved the Note. Yes, adding another constructor would be feasible. I would let this task for another PR though.